sycl/source/detail/common.cpp

```diff
@@ -12,6 +12,47 @@
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
+/// @brief CodeLocation information slot in thread local storage
+/// @details This structure is maintained by the SYCL runtime to manage the
+/// propagation of the code_location data down the stack without breaking ABI
+/// compatibility. This is used by the tls_code_loc_t class that is the
+/// prescribed way to propagate the code location information
+static thread_local detail::code_location GCodeLocTLS = {};
+
+/// @brief Default constructor to use in lower levels of the calling stack to
+/// check and see if code location object is available. If not, continue with
+/// instrumentation as needed
+tls_code_loc_t::tls_code_loc_t() {
+  // Check TLS to see if a previously stashed code_location object is
+  // available; if so, we are in a local scope.
+  MLocalScope = GCodeLocTLS.fileName() && GCodeLocTLS.functionName();
+}
+
+/// @brief Constructor to use at the top level of the calling stack
+/// @details This is usually a SYCL entry point used by the end user in their
+/// application code. In this case, we still checl to see if another code
+/// location has been stached in the TLS at a higher level. If not, we have the
+/// code location information that must be active for the current calling scope.
```

Comment:
Whether we have code location or not, the notifications are valid. If we have a TLS from a higher scope, then the object will pick that up instead of the any local code location information.